### PR TITLE
Add azure deployments + an example/tutorial for using Azure endpoints.

### PR DESCRIPTION
**Pull request by sorinsuciu-msft**

### CODE REVIEW

**Comment by ChrisHMSFT** [comment]
Hi @kennyhsu5 -- are you able to help review this PR? Thanks!

**Review by kennyhsu5 — APPROVED** [review]

File: openai/api_resources/deployment.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from openai.api_resources.abstract import DeletableAPIResource, ListableAPIResource, CreateableAPIResource
+from openai.error import InvalidRequestError
+
+
+class Deployment(CreateableAPIResource, ListableAPIResource, DeletableAPIResource):
```
[the remark this answers is not in the document]
Comment (sorinsuciu-msft) [reply openai/api_resources/deployment.py:6]: I added errors when calling the deployments operations if the type is not Azure

**Comment by sorinsuciu-msft** [comment]
@kennyhsu5 Can you please also merge it? I do not have permissions

**Comment by kennyhsu5** [comment]
> Can you please also merge it? I do not have permissions

Yea I will merge it once the error message is fixed.

**Pull request merged**